File: web/src/common/notify.js
const TYPES = {
  success: 'success',
  info: 'info',
  warning: 'warning',
  error: 'error',
}

export const Notify = (type, message, duration) => {
  if (window.$notify) {
    const t = TYPES[type.toLowerCase()]
    if (t) {
      window.$notify[t](message, { duration: duration })
      return
    }
  }
  console.log('Could not send notification')
}

File: client/notify.lua
UI.Notify = function(data)
  if type(data.type) ~= 'string' or type(data.message) ~= 'string' then
    error('UI: Missing required parameters for notify')
    return false
  end

  SendNUIMessage({
    action = 'notify',
    type = data.type,
    message = data.message,
    duration = data.duration,
  })
end
